tiledb/_array.py

```diff
@@ -0,0 +1,22 @@
+from typing import TYPE_CHECKING
+
+import tiledb.cc as lt
+
+from .ctx import CtxMixin
+
+if TYPE_CHECKING:
+    from libtiledb import Ctx
+
+
+class ArrayImpl(CtxMixin, lt.Array):
+    """
+    Represents a TileDB array.
+    """
```

Comment:
1. What is the purpose of this class, vs the Cython `Array`? It seems it just wraps an `Array` with a `Ctx`, but `ctx` is also passed to the `super().__init__` of `Subarray` so I'm not clear why it is needed at all. 

2. Currently this is used only by `Subarray`. Unless its use is to be expanded in the future, I'd move it to `subarray.py` (assuming it is actually needed).

3. Purpose and location aside, the name `ArrayImpl` seems misleading. Unlike `DenseArrayImpl` & `SparseArrayImpl`, this doesn't implement anything. Perhaps `ArrayWrapper`?

Reply:
`ArrayWrapper` is a better name. Long-story short: I need a C++ Array object to create a C++ Subarray object. This is a placeholder until we convert the rest of the Cython to PyBind. Depending on the order of conversion, it may get used for other PyBind classes or may be subsumed into the PyBind Array class (classes?) when they are fully implemented.

If you have a better way to do this, I am open to alternatives.

Reply:
You can just instantiate directly the underlying C++ Array; all `CtxMixin` is buying you here is the `default_ctx()` call if `ctx` is None, which you can do in one line:
```diff
diff --git a/tiledb/subarray.py b/tiledb/subarray.py
index 9b77939..b0071fa 100644
--- a/tiledb/subarray.py
+++ b/tiledb/subarray.py
@@ -5,8 +5,7 @@ import numpy as np
 
 import tiledb.cc as lt
 
-from ._array import ArrayWrapper
-from .ctx import Ctx, CtxMixin
+from .ctx import Ctx, CtxMixin, default_ctx
 from .libtiledb import Array
 
 Scalar = Real
@@ -18,18 +17,13 @@ class Subarray(CtxMixin, lt.Subarray):
     Represents a TileDB subarray.
     """
 
-    def __init__(
-        self,
-        array: Array,
-        ctx: Ctx = None,
-    ):
+    def __init__(self, array: Array, ctx: Ctx = None):
         """Class representing a subarray of a TileDB Array.
 
         :param array: tiledb.Array the subarray is defined on
         :param ctx: A TileDB Context
         """
-        self._array = array
-        super().__init__(ctx, ArrayWrapper(array))
+        super().__init__(ctx, lt.Array(ctx or default_ctx(), array))
 
     def add_dim_range(self, dim_idx: int, dim_range: Range):
         """Add a range to a dimension of the subarray.
```

Reply:
Done.